src/Enchantments.h

```diff
@@ -103,6 +104,9 @@ class cEnchantments
 	/** Returns true if there are no enchantments */
 	bool IsEmpty(void) const;
 
+	/** Returns true if all of our enchantments are compatible with the specified enchantment */
+	bool IsCompatibleWith(int a_EnchantmentID) const;
```

Comment:
CanAddEnchantment is a better name, and it also removes the ambiguity of if the enchantment is present (because you can add an enchantment to an item already with that enchantment).